omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskActionController1_8Test.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs1_8;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.module.webservices.helper.TaskAction;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.MockTaskFactoryWrapper;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskActionResource1_8;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskDefinitionResource1_8;
+import org.openmrs.scheduler.TaskDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNot.not;
+
+public class TaskActionController1_8Test extends MainResourceControllerTest {
+	
+	@Autowired
+	RestService restService;
+	
+	private TaskDefinition TestTask = new TaskDefinition(1, "TestTask", "TestTask Description",
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	MockTaskFactoryWrapper mockTaskFactory = new MockTaskFactoryWrapper();
+	
+	String TestTaskMockJson = "{\"id\" : 1,\"name\":\"TestTask\",\"description\":\"TestTask Description\","
+	        + " \"taskClass\":\"org.openmrs.scheduler.tasks.TestTask\"}";
+	
+	@Before
+	public void setUp() throws Exception {
+		mockTaskFactory.registeredTasks.add(TestTask.getName());
+		
+		TaskActionResource1_8 taskActionResource = (TaskActionResource1_8) restService
+		        .getResourceBySupportedClass(TaskAction.class);
+		taskActionResource.setTaskFactoryWrapper(mockTaskFactory);
+		
+		TaskDefinitionResource1_8 taskResource = (TaskDefinitionResource1_8) restService
+		        .getResourceBySupportedClass(TaskDefinition.class);
+		taskResource.setTaskFactoryWrapper(mockTaskFactory);
+	}
+	
+	@Test
+	public void shouldStartTestTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskFactory.scheduledTasks, not(hasItem(TestTask.getName())));
+		SimpleObject a = deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":["
+		        + TestTaskMockJson + "]}")));
+		assertThat(mockTaskFactory.scheduledTasks, hasItem(TestTask.getName()));
+	}
+	
+	@Test
+	public void shouldStopTestTask() throws Exception {
+		System.out.println("shouldStopTestTask");
```

Comment:
Take a look at this: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotprintout